File: csharp/1450.number-of-students-doing-homework-at-a-given-time.cs
/*
 * @lc app=leetcode id=1450 lang=csharp
 *
 * [1450] Number of Students Doing Homework at a Given Time
 */

using System.Linq;

// @lc code=start
public class Solution {
    public int BusyStudent(int[] startTime, int[] endTime, int queryTime) {
        return startTime
            .Zip(endTime, (s, e) => s <= queryTime && e >= queryTime)
            .Count(d => d);
    }
}
// @lc code=end

File: java/1450.number-of-students-doing-homework-at-a-given-time.java
/*
 * @lc app=leetcode id=1450 lang=java
 *
 * [1450] Number of Students Doing Homework at a Given Time
 */

// @lc code=start
class Solution {
    public int busyStudent(int[] startTime, int[] endTime, int queryTime) {
        int ret = 0, n = startTime.length;
        for (int i = 0; i < n; ++i) {
            if (startTime[i] <= queryTime && queryTime <= endTime[i]) {
                ++ret;
            }
        }
        return ret;
    }
}
// @lc code=end

